src/NuGet.Core/NuGet.Configuration/Settings/SettingItem.cs

```diff
@@ -101,5 +103,11 @@ internal virtual void Update(SettingItem setting)
                 AddOrUpdateAttribute(attribute.Key, attribute.Value);
             }
         }
+
+        public IEnumerable<XAttribute> GetXElementAttributes()
```

Comment:
Similar to one of JP's other comments, this class is designed to encapsulate the underlying XML file, and provide a higher level abstraction. So, to me, adding this as a public API feels like [a leaky abstraction](https://en.wikipedia.org/wiki/Leaky_abstraction).

On one hand, the way the settings classes are currently designed, it's supposed to hide the underlying file format, so exposing the internals violates the design.

On the other hand, if `dotnet nuget config` can't use the settings classes, that feels like there might be a design flaw too. Although in this case, you're using a "low level" API within NuGet.Configuration (SettingsItem), rather than the section specific abstractions that basically everything else is intended to use (PackageSource, PackageSourceMapping, etc).

Perhaps adding this as public API is the least bad option. It's not immediately clear to me what to recommend, so I'm not sure if this comment is actually helpful in any way. At least it's further evidence that API design is hard.